babashka-http-client/test/martian/babashka/http_client_test.clj

```diff
@@ -0,0 +1,152 @@
+              :body {:id 123}}
+             (select-keys response [:status :body]))))
+
+    (let [response @(martian/response-for m :get-pet {:id 123})]
+      (is (= {:name "Doggy McDogFace"
+              :type "Dog"
+              :age 3}
+             (:body response))))))
+(deftest error-handling-test
+  (testing "remote exceptions"
+
+    (testing "can be thrown"
+      (let [m (martian-http/bootstrap-swagger swagger-url {:interceptors martian-http/default-interceptors-async})]
+        (is (thrown? Exception @(martian/response-for m :get-pet {:id -1})))))
+
+    (testing "can be handled"
+      (let [turn-off-exception-throwing {:name ::turn-off-exception-throwing
+                                         :enter (fn [ctx] (assoc-in ctx [:request :throw] false))}
+            m (martian-http/bootstrap-swagger swagger-url {:interceptors (cons turn-off-exception-throwing
+                                                                               martian-http/default-interceptors-async)})]
+        (is (= 404 (:status @(martian/response-for m :get-pet {:id -1}))))))
+
+    (testing "can be caught in the interceptor chain"
+      (let [m (martian-http/bootstrap-swagger swagger-url {:interceptors (cons {:name ::i-catch-errors
+                                                                                :error (fn [ctx _error]
+                                                                                         (assoc ctx :response {:custom-error-response true}))}
+                                                                               martian-http/default-interceptors-async)})]
+        (is (= {:custom-error-response true}
+               @(martian/response-for m :get-pet {:id -1})))))))
+
+(deftest openapi-bootstrap-test
+  (let [m (martian-http/bootstrap-openapi openapi-url)
+        myaml (martian-http/bootstrap-openapi openapi-yaml-url)]
+
+    (is (= "https://sandbox.example.com"
+           (:api-root (martian-http/bootstrap-openapi openapi-test-yaml-url)))
+        "check yaml description")
+
+    (is (= "https://sandbox.example.com"
+           (:api-root (martian-http/bootstrap-openapi openapi-test-url)))
+        "check absolute server url")
+
+    (is (= "https://sandbox.com"
+           (:api-root (martian-http/bootstrap-openapi openapi-test-url {:server-url "https://sandbox.com"})))
+        "check absolute server url via opts")
+
+    (is (= "http://localhost:8888/v3.1"
+           (:api-root (martian-http/bootstrap-openapi openapi-test-url {:server-url "/v3.1"})))
+        "check relative server url via opts")
+
+    (is (= "http://localhost:8888/openapi/v3"
+           (:api-root myaml))
+        "check yaml description")
+
+    (is (contains? (set (map first (martian/explore myaml)))
+                   :get-order-by-id))
+
+    (is (= "http://localhost:8888/openapi/v3"
+           (:api-root m)))
+
+    (is (contains? (set (map first (martian/explore m)))
+                   :get-order-by-id))))
+
+(if-bb
```

Comment:
If I understand correctly, you let `lein test` start the server and then this will shell out to bb to run the bb tests while the server is still running - is that right?

And the tests for this module are run not by `lein sub` from the parent project, but instead by the extra addition to the circle config?

Reply:
> is that right?

This is right.

The tests here should be run like any of the other modules.

> instead by the extra addition to the circle config 

This was only to run tests for the core module.

Reply:
So I think I forgot to configure something for `lein sub`?

Reply:
For the record, I have no idea what lein sub is

Reply:
I actually mean `lein-modules` (I used to use `lein-sub` but not anymore). lein-modules is a plugin for lein that lets you have multiple child lein projects which can share some configuration around versioning and running tasks - e.g. running `lein install` from the top directory will run `lein install` on each child project (in the right order to satisfy dependencies).
https://github.com/jcrossley3/lein-modules

But in this case this new module already supports it with the inclusion of `:plugins [[lein-modules "0.3.11"]]` in the project.clj so I think we are all good here.